### ISSUE TITLE
Process multiples input lines

### ISSUE THREAD

**Issue by uggyuggy** [atc0005/safelinks#78]
Hi,

Thank's for sharing this tool.

Adding below few ideas/feature requests. As they are quite "close" and/or linked to each other (I think), I list those in a single "issue". 

It would be nice to have an option for "direct" results without extra informations lines and/or that even could be the default ouput and the current output with some kind of `-interactive`

```
$ usl 'https://emea01.safelinks.protection.outlook.com/?url=http%xxxxxxxxx'
example.net
```

Current output could also be
```
$ usl -interactive

Enter URL: https://emea01.safelinks.protection.outlook.com/?url=http%xxxxx
Original URL: http://example.net
```

It would be nice to be able to be able to bulk process multiple links at once in parallel

- From piped data

```
$ cat file.with.links | usl
http://example.com
http://example.net
```

- From input file

```
$ usl -inputfile file.with.links
http://example.com
http://example.net
```

**Comment by atc0005** [comment]
Overall I like the ideas, but I'd like to keep the current behavior of prompting the user if no input is provided. When I had others test the tool this behavior seemed intuitive to them and did not require further explanation.

Based on that, I think this would be the result:

- retain interactive prompt if no input given
- emit results without leading explanation text (i.e., drop `Original URL:` text and following newline)
- if input given, process and emit results without decoration until no further input provided (e.g., via pipe)
- if input file given, process the same way as stdin

As I mentioned I have other items ahead of this, so would probably work on this in pieces.

**Comment by atc0005** [comment]
@uggyuggy Dev build available here:

- https://github.com/atc0005/safelinks/releases/tag/v0.3.0-alpha.1

No doc updates yet, but hopefully the help text covers new behavior well enough.

**Comment by atc0005** [comment]
> No doc updates yet, but hopefully the help text covers new behavior well enough.

Updated README coverage:

- https://github.com/atc0005/safelinks/blob/v0.3.0-alpha.3/README.md

**Comment by atc0005** [comment]
Changes included in v0.3.0 release:

- https://github.com/atc0005/safelinks/releases/tag/v0.3.0

**Comment by uggyuggy** [comment]
Hi @atc0005 

Tested it, looks great.
Thank's  👍
